kafka/consumer.go

```diff
@@ -420,6 +491,10 @@ func (c *Consumer) ReadMessage(timeout time.Duration) (*Message, error) {
 // Close Consumer instance.
 // The object is no longer usable after this call.
 func (c *Consumer) Close() (err error) {
+	currentValue := atomic.LoadUint32(&c.isClosed)
```

Comment:
This line is not needed, just leave below:

```suggestion
	if !atomic.CompareAndSwapUint32(&c.isClosed, 0, 1) {
		return getOperationNotAllowedErrorForClosedClient()
	}
```

if isClosed is already 1 it'll return false